src/ts_type.rs

```diff
@@ -25,21 +25,23 @@ use std::fmt::Display;
 use std::fmt::Formatter;
 use std::fmt::Result as FmtResult;
 
-impl From<&TsLitType> for TsTypeDef {
-  fn from(other: &TsLitType) -> TsTypeDef {
+impl From<(&TsLitType, &ParsedSource)> for TsTypeDef {
```

Comment:
This code is extremely hard to follow. We probably shouldn't be using a `From` implementation on a tuple.